app/lib/delivery_failure_tracker.rb

```diff
@@ -44,6 +52,23 @@ def available?(url)
     def reset!(url)
       new(url).reset!
     end
+
+    def warning_domains
+      domains = Redis.current.keys(exhausted_deliveries_key('*')).map do |key|
+        key.delete_prefix(exhausted_deliveries_key(''))
+      end
+      domains - UnavailableDomain.all.pluck(:domain)
+    end
+
+    def warning_domains_map
+      warning_domains.each_with_object({}) { |domain, hash| hash[domain] = Redis.current.scard(exhausted_deliveries_key(domain)) }
+    end
+
+    private
+
+    def exhausted_deliveries_key(host)
+      "exhausted_deliveries:#{host}"
+    end
```

Comment:
Hmmm I'm unconfortable with the class method having the same name as the instance method… does that even work?

Reply:
Still a bit uneasy with this but I guess it is fine.